sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -53,7 +53,7 @@ static const pinmux_output_t kPinmuxOutputs[] = {
     /**
      * UART
      */
-    {.pad = kTopEarlgreyPinmuxMioOutIoc11,
+    {.pad = kTopEarlgreyPinmuxMioOutIoc4,
```

Comment:
The DV testbench for the ASIC target currently uses the same pinmux.c file, so we need to make the corresponding changes in that testbench as well:

https://github.com/lowRISC/opentitan/blob/01234f94e2cf1ef42f3e2e79e940b4ae5b199ecc/hw/top_earlgrey/dv/tb/tb.sv#L140-L141

Since the DFT strap signals are mapped to the same pins on the ASIC, you probably have to create a wire for each pin and attach both the `dft_straps` and `uart_rx/tx` interface signals to them (the straps are only used in DFT mode right at chip startup). Tagging @weicaiyang for advice.

Reply:
for this last part, we had a discussion with @timothytrippel some time back about possibly making this more flexible.
For practical purposes though, I would just make DV match the most likely use case for now to avoid too much overhead.

Reply:
Yes, tracking the (longer-term) feature request for flexible pinmux configs in #8380.

Reply:
Thanks @msfschaffner for considering the DV case. 
it sounds good to create a wire to switch these 2 pins to uart_rx/tx once the DFT select window is over. Can we figure out when to switch easily?

Reply:
Thank you all for the revew!
Would it be too bad swaping the pins IOC3/IOC4 with IOC10/IOC11? I mean map the `uart_tx/rx' to IOC3/IOC4 and the `dft_traps` to IOC10/IOC11.

Reply:
The DFT straps have to be in the same location, since I believe this has been spec'd as such in the pinout.

For now, let's code this up with wires like this:
```systemverilog
assign uart0_sel = 1'b1;
assign ioc3 = (uart0_sel) ? uart_if[0].uart_rx : dft_straps[0];
assign ioc4 = (uart0_sel) ? 1'bz : dft_straps[1];
assign uart_if[0].uart_tx = ioc4;
assign (weak0, weak1) ioc3 = 1'b1;
assign (weak0, weak1) ioc4 = 1'b1;
```
Note: in order to do so, the loop here needs to be adjusted
https://github.com/lowRISC/opentitan/blob/01234f94e2cf1ef42f3e2e79e940b4ae5b199ecc/hw/top_earlgrey/dv/tb/tb.sv#L287-L299

Also, the tie_off signals need to be redistributed after this change, since we now get two additional unassigned ports.